### PR TITLE
Patching Functional Tests

### PR DESCRIPTION
**Pull request by lpgeiger**
- Change the core system address from A550C18 to 0x0 (same for treasury etc.)

TODO: 
- functional tests can't set any amount of GAS besides 1GAS.

### CODE REVIEW

File: language/diem-framework/modules/0L/FIFO_rename.move [hunk 1]
```diff
@@ -6,17 +6,21 @@ address 0x1 {
 module FIFO {
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/FIFO_rename.move:6]: Any reason we renamed this file?

File: language/diem-framework/modules/0L/FIFO_rename.move [hunk 2]
```diff
@@ -6,17 +6,21 @@ address 0x1 {
...
     use 0x1::Vector;
 
-
-    struct FIFO<Element> has store {
+    struct FIFO<Element> has store, drop {
         incoming: vector<Element>,
         outgoing: vector<Element>
     }
+
+    public fun test() {
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/FIFO_rename.move:14]: Can we delete this?

File: language/diem-framework/modules/CoreAddresses.move [hunk 3]
```diff
@@ -14,22 +14,22 @@ module CoreAddresses {
     /// ultimate authority over the permissions granted (or removed) from
     /// accounts on-chain.
     public fun DIEM_ROOT_ADDRESS(): address {
-        0xA550C18
```
Comment (simsekgokhan) [inline language/diem-framework/modules/CoreAddresses.move:17]: Shouldn't we add /////// 0L ///////// here and all other similar places?

File: types/src/account_config/constants/addresses.rs [hunk 4]
```diff
@@ -6,7 +6,7 @@ use crate::account_address::AccountAddress;
 pub use move_core_types::language_storage::CORE_CODE_ADDRESS;
 
 pub fn diem_root_address() -> AccountAddress {
-    AccountAddress::from_hex_literal("0xA550C18")
+    AccountAddress::from_hex_literal("0x0")
```
Comment (simsekgokhan) [inline types/src/account_config/constants/addresses.rs:9]: /////// 0L ///////// here?

**Comment by simsekgokhan** [comment]
One more, when I build stdlib, I have this diff (which I should not, should be deterministic): 

```
. modified:
      032_FIFO.mv
      038_DiemAccount.mv
      FIFO_rename.md
```

Did you use `ol/util/build-stdlib.sh` to build stdlib? 
Or if you forgot to build stdlib after move changes, that is more likely be the issue.

File: language/diem-framework/modules/0L/FIFO_rename.move (hunk 1 above)
Comment (lpgeiger) [reply language/diem-framework/modules/0L/FIFO_rename.move:6]: I'll revert, this was a test.

File: language/diem-framework/modules/0L/FIFO_rename.move (hunk 2 above)
Comment (lpgeiger) [reply language/diem-framework/modules/0L/FIFO_rename.move:14]: yes

**Comment by lpgeiger** [comment]
@simsekgokhan Ok, I applied those changes suggested above

File: language/diem-framework/modules/0L/Oracle.move [hunk 5]
```diff
@@ -86,7 +88,7 @@ address 0x1 {
   
         // check if qualifies as a new round
         let is_new_round = current_height > upgrade_oracle.vote_window;
-  
+        print(&current_height);
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/Oracle.move:91]: Do we need this?

File: secure/key-manager/src/tests.rs [hunk 6]
```diff
@@ -748,7 +748,7 @@ fn verify_validator_config_info_mismatch<T: DiemInterface>(mut node: Node<T>) {
         script,
         MAX_GAS_AMOUNT,
         GAS_UNIT_PRICE,
-        XDX_NAME.to_owned(),
```
Comment (simsekgokhan) [inline secure/key-manager/src/tests.rs:751]: /////// 0L ///////// here?

File: language/move-lang/functional-tests/tests/0L/autopay/autopay_instruction_types.move [hunk 7]
```diff
@@ -58,10 +75,13 @@ use 0x1::GAS::GAS;
 script {
 use 0x1::AccountLimits;
 use 0x1::GAS::GAS;
+use 0x1::DiemAccount;
+use {{default}}::Holder;
   fun main(lr: signer, lucy_account: signer) {
       AccountLimits::publish_unrestricted_limits<GAS>(&lucy_account);
       AccountLimits::update_limits_definition<GAS>(&lr, {{lucy}}, 0, 10000, 0, 1);
       AccountLimits::publish_window<GAS>(&lr, &lucy_account, {{lucy}});
+    Holder::hold(&lucy_account, DiemAccount::balance<GAS>({{lucy}}));
```
Comment (simsekgokhan) [inline language/move-lang/functional-tests/tests/0L/autopay/autopay_instruction_types.move:84]: Minor alignment issue

**Pull request merged**